v2/protocol/http/protocol.go

```diff
@@ -201,7 +201,12 @@ func (p *Protocol) Respond(ctx context.Context) (binding.Message, protocol.Respo
 		if !ok {
 			return nil, nil, io.EOF
 		}
+
+		if in.msg == nil {
+			return nil, in.respFn, in.err
+		}
```

Comment:
An alternative would be to use reflection to compare `message` to `nil` here: https://github.com/cloudevents/sdk-go/blob/0fc16e13d87f0f2b1c99ec41183e35c0e68b6b5c/v2/binding/to_event.go#L23-L26

I believe we should avoid using reflection whenever possible.

Reply:
Fully agree!

Reply:
we use reflection in the client invoker but that is only processed once at startup.